orca-core/src/main/java/com/netflix/spinnaker/orca/config/TaskOverrideConfigurationProperties.kt

```diff
@@ -0,0 +1,46 @@
+package com.netflix.spinnaker.orca.config;
+
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.context.properties.ConstructorBinding
+
+/**
+ * Task overrides to use while planning stages.
+ */
+@ConfigurationProperties("task-overrides")
+@ConstructorBinding
+public class TaskOverrideConfigurationProperties(
+  /**
+   * list of task overrides.
+   */
+  public var overrideDefinitions: List<TaskOverrideDefinition> = listOf()
+) {
+
+  @ConstructorBinding
+  public class TaskOverrideDefinition(
+    /**
+     * Candidate stage in which we are looking to replace task definition
+     */
+    public var stageName: String,
+
+    /**
+     * Attribute values to consider in the stage context for overriding the task.
+     * For eg: this task override is only applicable for a particular cloud provider.
+     */
+    public var overrideCriteriaAttributes: List<String>,
+
+    /**
+     * Original task name as given while building the stage task graph
+     */
+    public var originalTaskName: String,
+
+    /**
+     * New task name to use in the task definition.
+     */
+    public var newTaskName: String,
```

Comment:
I think we should leave the task names alone and just replace the implementation class name.